cypress/integration/granules_spec.js

```diff
@@ -107,16 +109,52 @@ describe('Dashboard Granules Page', () => {
     });
 
     it('should display a link to download the granule list', () => {
-      cy.visit('/');
-
-      cy.contains('nav li a', 'Granules').as('granules');
-      cy.get('@granules').should('have.attr', 'href', '#/granules');
-      cy.get('@granules').click();
+      visitGranulePage();
 
       cy.contains('.heading--xlarge', 'Granules');
 
       cy.contains('a', 'Download Granule List')
       .should('have.attr', 'href').should('include', 'blob:http://');
     });
+
+    it('Should update dropdown with label when visiting bookmarkable URL', () => {
+      cy.visit('/#/granules?status=running');
+      cy.get('#form-Status-status > div > input').as('status-input');
+      cy.get('@status-input').should('have.value', 'Running');
+
+      cy.visit('/#/granules?status=completed');
+      cy.get('#form-Status-status > div > input').as('status-input');
+      cy.get('@status-input').should('have.value', 'Completed');
+    });
+
+    it('Should update dropdown with status value if query params are not a valid status.', () => {
+      cy.visit('/#/granules?status=notanoption');
+      cy.get('#form-Status-status > div > input').as('status-input');
+      cy.get('@status-input').should('have.value', 'notanoption');
+    });
+
```

Comment:
Hmm, is this a behavior that we want? That entering arbitrary values in the URL actually show up in the component?

Reply:
It would be the equivalent if you typed something in the dropdown box that didn't match any of the configured options.  I don't know if we want to prevent that or not?

Ok, I just actually tested that. and if you type a `z` in the dropdown and it doesn't match anything, nothing is kicked off, and if you hit return (select), the url pushes a ? before the hash so we get `http://localhost:3000/?#/granules` . That doesn't seem right.  

I'll have to figure out how to handle this with search params differently from dropdowns.  But Ok I can see this might not be what we want.

To clarify:  you think that if you browse to a url that has a invalid status key, the dropdown corresponding to that paramKey should clear itself, and remove that from the query?

so basically L133 should be `      cy.get('@status-input').should.not('have.value', 'notanoption');`

Reply:
I think if the URL has an invalid status key, basically nothing should happen, so whatever logic initially sets the value should short-circuit if the key is invalid.

We probably also don't want the case you described where typing a invalid value in a dropdown and hitting enter affects the URL

Reply:
What should the state of the URL be after discovering a bad value?
say `?status=badvalue`, should that be removed to match the dropdown value that is also blank?

In the case of the problem with the URL updating badly, that's an existing bug that I filed a ticket for https://bugs.earthdata.nasa.gov/browse/CUMULUS-1621, but I'm hoping that it will go away in an update to react router or an update to a browser history instead of a hash history.